lti_consumer/plugin/views.py

```diff
@@ -416,3 +425,196 @@ def scores(self, request, *args, **kwargs):  # pylint: disable=unused-argument
+        user_ids = [user['id'] for user in data]
+        users = get_user_model().objects.filter(id__in=user_ids)
+        external_ids = compat.batch_get_or_create_externalids(users)
+
+        for user in data:
+            # append external ids to user dictonary
+            user['external_id'] = external_ids[user['id']].external_user_id
+
+            # if user's image exists turn it to an absolute url
+            if user['profile']['profile_image']['has_image']:
+                picture_url = user['profile']['profile_image']['image_url_small']
+                user['profile']['profile_image']['image_url_small'] = self.request.build_absolute_uri(picture_url)
+            processed_data.append(user)
+
+        return processed_data
+
+    def get_serializer(self, data, **kwargs):  # pylint: disable=arguments-differ
+        """
+        Override get_serializer method to change data structure
+        to match LtiNrpsContextMembershipSerializer.
+        """
+
+        # prepare datastructure supported by serializer
+        result = {
+            'id': self.request.build_absolute_uri(),
+            'context': {
+                'id': self.request.lti_configuration.location.course_key
+            },
+            'members': self.preprocess(data),
+        }
+        return super().get_serializer(result)
+
+    def get_serializer_context(self):
+        """
+        Adds additional contexts for serializers. Ex: if PII fileds can be exposed.
+        """
+        context = super().get_serializer_context()
+        context.update({
+            'expose_pii_fields': expose_pii_fields(self.request.lti_configuration.location.course_key)
+        })
+        return context
+
+    def get_serializer_class(self):
+        """
+        Overrides ModelViewSet's `get_serializer_class` method.
+        Checks if PII fields can be exposed and returns appropiate serializer.
+        """
+
+        if expose_pii_fields(self.request.lti_configuration.location.course_key):
+            return LtiNrpsContextMembershipPIISerializer
+        else:
+            return LtiNrpsContextMembershipBasicSerializer
+
+    def list(self, *args, **kwargs):  # pylint: disable=unused-argument
+        """
+        Overrides default list method of ModelViewSet.
+        Checks if enrollment limit satisfies for current request.
+        """
+        # paginator gets request object via `paginate_queryset` call, since we are not using that
+        # we need to set the request object for paginator ourselves.
+        self.paginator.request = self.request
+
+        # get course key
+        course_key = self.request.lti_configuration.location.course_key
+
+        # parse and build filter parameters for `get_course_member` python API
+        nrps_filter_role = self.request.query_params.get('role', None)
+        filter_by = self.parse_role_filter(nrps_filter_role=nrps_filter_role)
+        access_roles, include_students = self.build_filter_kwargs(filter_by=filter_by)
```

Comment:
These steps do not need to happen here. See this + my previous suggestions on the static methods:
```suggestion
        # Parse and build filter parameters for `get_course_member` python API
        access_roles, include_students = self.get_filter_params(
            self.request.query_params.get('role', None)
        )
```

Reply:
Simplified, but not exact. Due to my previous comment [here](https://github.com/edx/xblock-lti-consumer/pull/124#discussion_r598113187).